### PR TITLE
refactor: add pauser role for redistribution contract

### PR DESCRIPTION
**Pull request by swarmHaseeb**
ESLinting in `Redistribution.test.ts`, tests for pausing contract and **Pausing Functionality** in `Redistribution.sol` contract.

### CODE REVIEW

**Review by metacertain — APPROVED** [review]
Thanks!

**Review by notanatol — APPROVED** [review]

File: src/Redistribution.sol [hunk 1]
```diff
@@ -97,6 +97,23 @@ contract Redistribution is AccessControl, Pausable {
     // The reveal of the winner of the last round.
     Reveal public winner;
 
+    /**
+    * @dev Pause the contract. The contract is provably stopped by renouncing
+     the pauser role and the admin role after pausing, can only be called by the `PAUSER`
+     */
+    function pause() public {
+        require(hasRole(PAUSER_ROLE, msg.sender), "only pauser can pause");
```
Comment (mrekucci) [inline src/Redistribution.sol:105]: I recommend staying consistent with the messages in the `PriceOracle.sol`: "caller is not pauser". The same is below.

**Review by mrekucci — APPROVED** [review]

File: test/Redistribution.test.ts [hunk 2]
```diff
@@ -543,7 +571,7 @@ describe('Redistribution', function () {
         const obsfucatedHash = encodeAndHash(overlay_2, depth_2, hash_2, reveal_nonce_2);
 
         const currentRound = await r_node_2.currentRound();
-        console.log(currentRound)
+        console.log(currentRound);
```
Comment (significance) [inline test/Redistribution.test.ts:574]: shall we remove this

**Review by significance — CHANGES_REQUESTED** [review]
other than minor things metnioned this lgtm, thanks guys! 😄

**Pull request merged**